backup.sh

```diff
@@ -0,0 +1,93 @@
+#!/bin/sh
+
+set -e -x
+
+export RUN_MODE=`basename "$0"`
+
+for i in "$@"
+do
+case $i in
+--cloud=*)
+  export CLOUD="${i#*=}"
+  shift
+  ;;
+--region=*)
+  export REGION="${i#*=}"
+  shift
+  ;;
+--bucket=*)
+  export BUCKET="${i#*=}"
+  shift
+  ;;
+--endpoint=*)
+  export ENDPOINT="${i#*=}"
+  shift
+  ;;
+--backup-dir=*|--srcDir=*)
+  export BACKUP_DIR="${i#*=}"
+  shift
+  ;;
+--destDir=*)
+  export DEST_DIR="${i#*=}"
+  shift
+  ;;
+  *)
+  common::log "unknown option [$i]"
+  exit 1
+  ;;
+esac
+
+done
+
+if [ "$CLOUD" = "gcp" ] && [ -z "$GOOGLE_APPLICATION_CREDENTIALS" ]; then
+  echo "GCP credentials path is not set"
+  exit 1
+fi
+
+if [ "$CLOUD" = "ceph" ] || [ "$CLOUD" = "aws" ]; then
+  if [ -z "$AWS_ACCESS_KEY_ID" ]; then
+    echo "S3 access key is not set"
+    exit 1
+  fi
+  if [ -z "$AWS_SECRET_ACCESS_KEY" ]; then
+    echo "S3 access secret is not set"
+    exit 1
+  fi
+fi
```

Comment:
Need to add a check for variable `$CLOUD`, if the variable `$CLOUD` is not any of gcp,aws,ceph, need to log the error info and exit program.